bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/AccessibleToolBar.java

```diff
@@ -0,0 +1,77 @@
+package org.eclipse.ui.texteditor;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.TraverseEvent;
+import org.eclipse.swt.events.TraverseListener;
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.ToolBar;
+import org.eclipse.swt.widgets.ToolItem;
+
+import org.eclipse.jface.layout.GridLayoutFactory;
+
+/**
+ * This class wraps the ToolBar to make it possible to use tabulator-keys to
+ * navigate between the buttons of a ToolBar. For this, we simulate a singular
+ * ToolBar by putting each ToolItem into it's own ToolBar and composing them
+ * into a Composite. Since the "Enter" keypress could not previously trigger
+ * activation behavior, we listen for it manually and send according events if
+ * necessary.
+ *
+ * @since 3.18
+ */
+class AccessibleToolBar extends Composite {
+
+	private List<ToolBar> toolBars = new ArrayList<>();
+
+	public AccessibleToolBar(Composite parent) {
+		super(parent, SWT.NONE);
+		GridLayoutFactory.fillDefaults().numColumns(0).spacing(0, 0).margins(0, 0).applyTo(this);
+	}
+
+	/**
+	 * Creates a ToolItem handled by this ToolBar and returns it. Will add a
+	 * KeyListener which will catch all "Enter"-Keypresses.
+	 *
+	 * @param styleBits the StyleBits to apply to the created ToolItem
+	 * @return a newly created ToolItem
+	 */
+	public ToolItem createToolItem(int styleBits) {
+		ToolBar parent = new ToolBar(this, SWT.FLAT | SWT.HORIZONTAL);
+		ToolItem result = new ToolItem(parent, styleBits);
+
+		addToolItemTraverseListener(parent, result);
+
+		((GridLayout) getLayout()).numColumns++;
+
+		toolBars.add(parent);
+		return result;
+	}
+
+	private void addToolItemTraverseListener(ToolBar parent, ToolItem result) {
+		parent.addTraverseListener(new TraverseListener() {
+
+			@Override
+			public void keyTraversed(TraverseEvent e) {
+				if (e.keyCode == SWT.CR || e.keyCode == SWT.KEYPAD_CR) {
+					result.setSelection(!result.getSelection());
+					e.doit = false;
+				}
+			}
+
+		});
+	}
+
+	@Override
+	public void setBackground(Color color) {
+		super.setBackground(color);
+		for (ToolBar bar : toolBars) { // we need to color each toolbar separately
```

Comment:
Why do we need to do it? Is it a bug? Which one?
Please add some more info to the comment or delete it. It doesn't  add any value the way it is right now.

Reply:
I don't believe this comment is too important - in short: the ToolBar ToolItems don't all (easily) inherit the color of their parent. In particular, the "separator" is a culprit!

![grafik](https://github.com/eclipse-platform/eclipse.platform.ui/assets/16443184/d8e9de01-993d-483c-9d8b-d0194e2b95fc)

Reply:
I will keep the comment so that nobody accidentally removes the code (which might seem redundant as usually properties are passed down regularly to their children) in the future

Reply:
Edited in https://github.com/eclipse-platform/eclipse.platform.ui/commit/0c0d22bc8f0ea13eb6e1124c89747f7a4f02e266
Note that the formatting problems that are displayed on gitub are not relevant as the formatting looks good in Eclipse:

![grafik](https://github.com/eclipse-platform/eclipse.platform.ui/assets/16443184/f9c1ea16-bf8e-4fe2-adc2-b3bd868b9d77)